From 41152d25933496ccd7c7011fed771389036db2f8 Mon Sep 17 00:00:00 2001
From: tummychow <tummychow@users.noreply.github.com>
Date: Wed, 30 Apr 2014 00:03:56 -0400
Subject: [PATCH] Implement preliminary CodeRay numbering support

This commit provides support for :line_numbers => :table in the CodeRay
filter. However, the settings from Poole are interfering with the
padding. If you turn on line numbering, there's way too much blank space
around the edges, unless you reduce the padding in the coderay.css pre
selector. But then, if you turn off line numbering, there's not enough
blank space.

The problem is ultimately with poole.css. I don't think I can find a
solution without changing it somehow, which is something I'd rather not
do.
---
 Rules                   |  2 +-
 content/css/coderay.css | 15 ++++++++++-----
 2 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/Rules b/Rules
index 381356f..7723132 100644
--- a/Rules
+++ b/Rules
@@ -71,7 +71,7 @@ compile '*' do
       filter :redcarpet, :renderer => HTMLPantsCode,
         :options => {:fenced_code_blocks => true, :tables => true, :strikethrough => true}
       filter :colorize_syntax, :default_colorizer => :coderay,
-        :coderay => {:line_numbers => nil, :line_number_anchors => false, :bold_every => false}
+        :coderay => {:line_numbers => :table, :line_number_anchors => false, :bold_every => false}
       snapshot :render_md # currently equivalent to :pre
     end
 
diff --git a/content/css/coderay.css b/content/css/coderay.css
index c4f3d50..84140c8 100644
--- a/content/css/coderay.css
+++ b/content/css/coderay.css
@@ -12,11 +12,17 @@
   overflow: auto;
   white-space: pre;
   word-wrap: normal;
+  padding: .7em 0 0;
+}
+
+.CodeRay .code {
+  width: 100%;
+  background-color: transparent;
 }
 
 span.CodeRay {
   white-space: pre;
-  padding: 2px;
+  padding: 0;
   border: none;
 }
 
@@ -28,12 +34,13 @@ span.CodeRay {
 
 table.CodeRay {
   width: 100%;
-  padding: 2px;
+  padding: 0;
   border-collapse: collapse;
 }
 
 table.CodeRay td {
-  padding: 2px 4px;
+  padding-right: .7em;
+  border: none;
   vertical-align: top;
 }
 
@@ -50,8 +57,6 @@ table.CodeRay td {
 
 .CodeRay span.line-numbers { padding: 0 4px; }
 
-.CodeRay .code { width: 100%; }
-
 .CodeRay .comment,
 .CodeRay .char .delimiter,
 .CodeRay .regexp .delimiter,
-- 
1.9.2

